src/app/store/store.ts

```diff
@@ -0,0 +1,13 @@
+'use client';
+
+import { configureStore } from '@reduxjs/toolkit';
+import oldboyReducer from './Features/OldboyBarbershop/OldBoySlice';
+
+export const store = configureStore({
+  reducer: {
+    oldboy: oldboyReducer,
+  },
+});
+
+export type RootState = ReturnType<typeof store.getState>;
+export type AppDispatch = typeof store.dispatch;
```

Comment:
You need to add custom hook instead enhancing the existing `useDispatch` hook because it does not handle types very well. See example here https://redux-toolkit.js.org/usage/usage-with-typescript#getting-the-dispatch-type

Reply:
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";